src/permission/fs_permission.h

```diff
@@ -0,0 +1,148 @@
+#ifndef SRC_PERMISSION_FS_PERMISSION_H_
+#define SRC_PERMISSION_FS_PERMISSION_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "v8.h"
+
+#include <vector>
+#include "permission/permission_base.h"
+
+namespace node {
+
+namespace permission {
+
+class FSPermission final : public PermissionBase {
+ public:
+  void Apply(const std::string& deny) override;
+  bool Deny(PermissionScope scope,
+            const std::vector<std::string>& params) override;
+  bool is_granted(PermissionScope perm, const std::string& param) override;
+
+  // For debugging purposes, use the gist function to print the whole tree
+  // https://gist.github.com/RafaelGSS/5b4f09c559a54f53f9b7c8c030744d19
```

Comment:
It seems like this debugging function should be in the codebase and not a gist.

Reply:
Yeah. Actually, I'm planning to open another PR once it lands to include something similar to `--debug-permission-model` that will show some interesting information. Are you ok doing that in a separate PR?

Reply:
Sure.